lib/modules/manager/maven/update.ts

```diff
@@ -78,7 +79,30 @@ export function bumpPackageVersion(
     const startTagPosition = versionNode.startTagPosition;
     const versionPosition = content.indexOf(versionNode.val, startTagPosition);
 
-    const newPomVersion = semver.inc(currentValue, bumpVersion);
+    let newPomVersion: string | null = null;
+    const currentPrereleaseValue = semver.prerelease(currentValue);
+    if (isSnapshot(currentPrereleaseValue)) {
+      // It is already a SNAPSHOT version.
+      // Therefore the same qualifier (prerelease) will be used as before.
+      let releaseType = bumpVersion;
+      if (!bumpVersion.startsWith('pre')) {
+        releaseType = ('pre' + bumpVersion) as ReleaseType;
+      }
+      newPomVersion = semver.inc(
+        currentValue,
+        releaseType,
+        currentPrereleaseValue?.join('.'),
+        false,
+      );
+    } else if (currentPrereleaseValue) {
+      // Some qualifier which is not a SNAPSHOT is present.
+      // The expected behaviour in this case is unclear and the standard increase will be used.
+      newPomVersion = semver.inc(currentValue, bumpVersion);
+    } else {
+      // A release version without any qualifier is present.
+      // Therefore the SNAPSHOT qualifier will be added if a prerelease is requested.
+      newPomVersion = semver.inc(currentValue, bumpVersion, 'SNAPSHOT', false);
```

Comment:
```suggestion
      // Therefore the SNAPSHOT qualifier will be added if a prerelease is requested.
      // This will do a normal increment, ignoring SNAPSNOT, if a non-prerelease bumpVersion is configured
      newPomVersion = semver.inc(currentValue, bumpVersion, 'SNAPSHOT', false);
```